freebsd/src/event.rs

```diff
@@ -149,6 +151,58 @@ impl std::future::Future for EventFdNotified<'_> {
     }
 }
 
+pub trait KqueueExt {
+    fn wait_events(&self, changelist: &[KEvent], eventlist: &mut [KEvent]) -> nix::Result<usize>;
+}
+
+pub fn kevent_classic(
+    kq: i32,
+    changelist: &[KEvent],
+    eventlist: &mut [KEvent],
+) -> nix::Result<usize> {
+    let mut interrupted = true;
+    while interrupted {
+        let res = unsafe {
+            nix::libc::kevent(
+                kq,
+                changelist.as_ptr() as *const nix::libc::kevent,
+                changelist.len() as i32,
+                eventlist.as_mut_ptr() as *mut nix::libc::kevent,
+                eventlist.len() as i32,
+                std::ptr::null(),
+            )
+        };
+
+        let errno = nix::errno::errno();
+
+        match res {
+            -1 => {
+                if errno != nix::libc::EINTR {
+                    return Err(nix::errno::Errno::EINTR);
```

Comment:
This should probably be `Err(errno)`, rather than returning the one error it can't be.

Reply:
That is indeed a typo